app/src/main/java/net/squanchy/schedule/tracksfilter/ScheduleTracksFilter.kt

```diff
@@ -28,43 +33,50 @@ class ScheduleTracksFilterActivity : AppCompatActivity() {
     private lateinit var trackAdapter: TracksFilterAdapter
 
     private var subscription: Disposable? = null
+    private var checkableTracks: List<CheckableTrack>? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_track_filters)
 
+        backgroundDim.setOnClickListener { finish() }
+        closeButton.setOnClickListener { finish() }
+
         val component = tracksFilterComponent(this)
         tracksRepository = component.tracksRepository()
         tracksFilter = component.tracksFilter()
 
-        trackAdapter = TracksFilterAdapter(this)
+        trackAdapter = TracksFilterAdapter(this) { track, selected ->
+            val selectedTracks = checkableTracks?.allSelected() ?: emptySet()
+            val newSelectedTracks = selectedTracks.addOrRemove(track, selected)
+            tracksFilter.updateSelectedTracks(newSelectedTracks)
+        }
 
-        trackFiltersList.layoutManager = LinearLayoutManager(this)
+        trackFiltersList.layoutManager = FlexboxLayoutManager(this, FlexDirection.ROW)
+        trackFiltersList.addItemDecoration(FlexboxItemDecoration(this).apply {
+            setDrawable(resources.getDrawable(R.drawable.filters_separator, theme))
```

Comment:
Any reason why we use `Resources.getDrawable` instead of `ContextCompat`? I guess the implementation is the same for API 21+

Reply:
Hm I only use ContextCompat for stuff that has been deprecated (or to get `getColor` on context which got it in 23 IIRC)